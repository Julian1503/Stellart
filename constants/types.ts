export type NavLink = {
  id: number;
  link: string;
  text: string;
}