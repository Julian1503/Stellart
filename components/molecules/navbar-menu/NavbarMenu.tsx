import { useState } from "react";
import CloseButton from "@/atoms/close-button/CloseButton";
import NavButton from "@/atoms/nav-button/NavButton";
import Button from "@/atoms/button/Button";
import classes from "./navbar-menu.module.css";
import { NavLink } from "@/constants/types";
import BurgerButton from "@/atoms/burger-button/BurgerButton";

type NavbarMenuProps = { links: Array<NavLink>; showLoginButton: boolean; isMenuOpen: boolean; handleMenu: ()=>void};

function NavbarMenu({ links, showLoginButton, isMenuOpen, handleMenu}: NavbarMenuProps) {
  return (
    <>
      <BurgerButton
        className={classes.btn_burger}
        onClick={handleMenu}
      />
      <div
        data-testid="menu"
        className={`${classes.links_container}  ${
          isMenuOpen ? classes.open : ""
        }`}
      >
        <CloseButton
          className={isMenuOpen ? classes.btn_close_menu_opened : classes.btn_close_menu_closed}
          onClick={handleMenu}
        />
        <ul>
          {links.map((link) => (
            <NavButton id={link.id} key={link.id} link={link.link} text={link.text}/>
          ))}
          {showLoginButton && (
            <li>
              <Button
                hasShadow
                width={170}
                height={49}
                link={"/Login"}
                text={"Login"}
              />
            </li>
          )}
        </ul>
      </div>
    </>
  );
}

export default NavbarMenu;
