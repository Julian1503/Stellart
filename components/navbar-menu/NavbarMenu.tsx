import { useState } from "react";
import CloseButton from "../close-button/CloseButton";
import NavButton from "../nav-button/NavButton";
import Button from "../button/Button";
import classes from "./navbar-menu.module.css";
import { NavLink } from "@/constants/types";
import BurguerButton from "../burguer-button/BurguerButton";

type NavbarMenuProps = { links: Array<NavLink>; showLoginButton: boolean; isMenuOpen: boolean; handleMenu: ()=>void};

function NavbarMenu({ links, showLoginButton, isMenuOpen, handleMenu}: NavbarMenuProps) {
  return (
    <>
      <BurguerButton
        className={classes.btn_burguer}
        onClick={handleMenu}
      />
      <div
        className={`${classes.links_container}  ${
          isMenuOpen ? classes.open : ""
        }`}
      >
        <CloseButton
          className={classes.btn_close}
          onClick={handleMenu}
        />
        <ul>
          {links.map((link) => (
            <NavButton key={link.id} link={link.link} text={link.text} />
          ))}
          {showLoginButton && (
            <li>
              <Button
                hasShadow
                width={170}
                height={49}
                link={"/Login"}
                text={"Login"}
              />
            </li>
          )}
        </ul>
      </div>
    </>
  );
}

export default NavbarMenu;
