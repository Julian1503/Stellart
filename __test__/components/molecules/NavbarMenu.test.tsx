import React, { useState } from "react";
import { render, fireEvent } from "@testing-library/react";
import { NavLink } from "@/constants/types";
import NavbarMenu from "@/components/molecules/navbar-menu/NavbarMenu";

const menuTestId = "menu";
const btnLoginTestId = "btn";

describe("Navbar Menu", () => {
  let links: Array<NavLink>,
    showLoginButton: boolean,
    isMenuOpen: boolean,
    handleMenu: () => void;

  beforeEach(() => {
    links = [
      { id: 1, link: "/home", text: "Home" },
      { id: 2, link: "/about", text: "About" },
      { id: 3, link: "/contact", text: "Contact" },
    ];
    showLoginButton = true;
    isMenuOpen = false;
    handleMenu = jest.fn();
  });

  it("should render the component with the correct links", () => {
    const { getByTestId } = render(
      <NavbarMenu
        links={links}
        showLoginButton={false}
        isMenuOpen={false}
        handleMenu={() => {}}
      />
    );
    console.log(getByTestId(menuTestId));
    const navLinks = getByTestId(menuTestId).querySelectorAll(
      '[data-testid^="btnNav"]'
    );

    expect(navLinks.length).toBe(links.length);
    links.forEach((link, index) => {
      expect(navLinks[index]).toHaveTextContent(link.text);
    });
  });

  it("should render the login button when showLoginButton is set to true", () => {
    const { getByTestId } = render(
      <NavbarMenu
        links={links}
        showLoginButton={true}
        isMenuOpen={false}
        handleMenu={() => {}}
      />
    );
    const loginBtn = getByTestId(btnLoginTestId);

    expect(loginBtn).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const button = render(
      <NavbarMenu
        links={links}
        showLoginButton={false}
        isMenuOpen={true}
        handleMenu={() => {}}
      />
    );
    expect(button).toMatchSnapshot();
  });
});
