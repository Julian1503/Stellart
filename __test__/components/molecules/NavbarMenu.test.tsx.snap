// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Menu matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        arial-label="Burger button"
        class="menu_btn btn_burger"
        data-testid="btnBurger"
      >
        <span
          class="burger"
        >
          <span
            class="line"
          />
          <span
            class="line"
          />
          <span
            class="line"
          />
        </span>
      </span>
      <div
        class="links_container  open"
        data-testid="menu"
      >
        <span
          class="btn_close btn_close_menu_opened"
          data-testid="btnClose"
        >
          <span
            class="line"
          />
          <span
            class="line"
          />
        </span>
        <ul>
          <li
            class="link "
            data-testid="btnNav1"
          >
            <a
              href="/home"
            >
              Home
            </a>
          </li>
          <li
            class="link "
            data-testid="btnNav2"
          >
            <a
              href="/about"
            >
              About
            </a>
          </li>
          <li
            class="link "
            data-testid="btnNav3"
          >
            <a
              href="/contact"
            >
              Contact
            </a>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <span
      arial-label="Burger button"
      class="menu_btn btn_burger"
      data-testid="btnBurger"
    >
      <span
        class="burger"
      >
        <span
          class="line"
        />
        <span
          class="line"
        />
        <span
          class="line"
        />
      </span>
    </span>
    <div
      class="links_container  open"
      data-testid="menu"
    >
      <span
        class="btn_close btn_close_menu_opened"
        data-testid="btnClose"
      >
        <span
          class="line"
        />
        <span
          class="line"
        />
      </span>
      <ul>
        <li
          class="link "
          data-testid="btnNav1"
        >
          <a
            href="/home"
          >
            Home
          </a>
        </li>
        <li
          class="link "
          data-testid="btnNav2"
        >
          <a
            href="/about"
          >
            About
          </a>
        </li>
        <li
          class="link "
          data-testid="btnNav3"
        >
          <a
            href="/contact"
          >
            Contact
          </a>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
